x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/CustomLinkFlyout/LinkPreview.tsx

```diff
@@ -17,28 +17,31 @@ import {
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { debounce } from 'lodash';
+import { Filter } from '../../../../../../../../../../plugins/apm/common/customLink/custom_link_types';
 import { Transaction } from '../../../../../../../../../../plugins/apm/typings/es_schemas/ui/transaction';
 import { callApmApi } from '../../../../../../services/rest/createCallApmApi';
-import {
-  FilterKeyValue,
-  convertFiltersToObject,
-  replaceTemplateVariables
-} from './helper';
+import { replaceTemplateVariables } from './helper';
 
 interface Props {
   label: string;
   url: string;
-  filters: FilterKeyValue[];
+  filters: Filter[];
 }
 
 const fetchTransaction = debounce(
-  async (
-    filters: FilterKeyValue[],
-    callback: (transaction: Transaction) => void
-  ) => {
+  async (filters: Filter[], callback: (transaction: Transaction) => void) => {
+    const query = filters.reduce(
+      (acc: Record<string, string>, { key, value }) => {
+        if (key && value) {
+          acc[key] = value;
+        }
+        return acc;
+      },
+      {}
+    );
```

Comment:
Do you need to do this? Can't you send the array as a query param? In `get_transaction` you convert this back to an array, so would be good if we didn't have to change the format just for transit.

Reply:
I tried, but this is the request when sending an array: `http://localhost:5601/api/apm/settings/custom_links/transaction?0=&1=`